tests/sentry/models/test_projecttemplateoption.py

```diff
@@ -3,6 +3,113 @@
 from sentry.testutils.cases import TestCase
 
 
+class ProjectTemplateOptionManagerTest(TestCase):
+    def setUp(self):
+        self.org = self.create_organization()
+
+        self.project_template = ProjectTemplate.objects.create(
+            name="test_project_template", organization=self.org
+        )
+
+    def test_set_value(self):
+        ProjectTemplateOption.objects.set_value(self.project_template, "foo", "bar")
+        assert (
+            ProjectTemplateOption.objects.get(
+                project_template=self.project_template, key="foo"
+            ).value
+            == "bar"
+        )
+
+    def test_get_value(self):
+        result = ProjectTemplateOption.objects.get_value(self.project_template, "foo")
+        assert result is None
+
+        ProjectTemplateOption.objects.create(
+            project_template=self.project_template, key="foo", value="bar"
+        )
+
+        result = ProjectTemplateOption.objects.get_value(self.project_template, "foo")
+        assert result == "bar"
+
+    def test_get_value_validated(self):
+        result = ProjectTemplateOption.objects.get_value(self.project_template, "foo")
+        assert result is None
+
+        ProjectTemplateOption.objects.create(
+            project_template=self.project_template, key="foo", value="bar"
+        )
+
+        # Test validator true
+        result = ProjectTemplateOption.objects.get_value(
+            self.project_template, "foo", None, lambda x: x == "bar"
+        )
+        assert result == "bar"
+
+        # Test validator false
+        result = ProjectTemplateOption.objects.get_value(
+            self.project_template, "foo", None, lambda x: x != "bar"
+        )
+        assert result is None
+
+    def test_unset_value(self):
+        ProjectTemplateOption.objects.unset_value(self.project_template, "foo")
```

Comment:
why do you unset it here?

Reply:
ah, whoops! this exposed an issue and i forgot to clean it up, will remove :)